src/geometry/manifold/charted_space.lean

```diff
@@ -365,6 +366,86 @@ instance : order_top (structure_groupoid H) :=
   le_top := λ u f hf, by { split; exact dec_trivial },
   ..structure_groupoid.partial_order }
 
+/-- A groupoid is closed under restriction if it contains all restrictions of its element local
+homeomorphisms to open subsets of the source. -/
+class closed_under_restriction (G : structure_groupoid H) : Prop :=
+(closed_under_restriction : ∀ {e : local_homeomorph H H}, e ∈ G.members → ∀ (s : set H), is_open s →
+  (e : local_homeomorph H H).restr s ∈ G.members)
```

Comment:
```suggestion
(closed_under_restriction : ∀ {e : local_homeomorph H H}, e ∈ G → ∀ (s : set H), is_open s →
  (e : local_homeomorph H H).restr s ∈ G)
```
There is a coercion, so that one should essentially never write `∈ G.members`, and only `∈ G` instead. If you make the switch, you may notice that some fields of structure groupoids (that don't use the coercion version) don't apply anymore, but they are reformulated below without primes and using the coercion, so these should be favored also.

Reply:
Thanks, this was instructive!